service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java

```diff
@@ -1,5 +1,64 @@
 package bio.terra.workspace.service.resource.controlled.cloud.azure;
 
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.common.exception.AzureManagementExceptionUtils;
 import bio.terra.workspace.service.resource.controlled.flight.delete.DeleteControlledResourceStep;
+import com.azure.core.management.exception.ManagementException;
+import java.util.Arrays;
+import java.util.List;
+import org.springframework.http.HttpStatus;
 
-public abstract class DeleteAzureControlledResourceStep implements DeleteControlledResourceStep {}
+public abstract class DeleteAzureControlledResourceStep implements DeleteControlledResourceStep {
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    try {
+      return deleteResource(context);
+    } catch (InterruptedException e) {
+      throw e;
```

Comment:
Why is this catching and rethrowing?

Reply:
This makes it so the next clause handles all exceptions _except_ if it's an `InterruptedException`.

I'll add a comment so it's clear in the future.